frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx

```diff
@@ -79,49 +95,95 @@ export default function CommunicationPreferencePage() {
     setDigitalMethodChecked(false);
   };
 
+  /**
+   * Gets an error message based on the provided internationalization (i18n) key.
+   *
+   * @param errorI18nKey - The i18n key for the error message.
+   * @returns The corresponding error message, or undefined if no key is provided.
+   */
+  function getErrorMessage(errorI18nKey?: string): string | undefined {
+    if (!errorI18nKey) return undefined;
+
+    /**
+     * The 'as any' is employed to circumvent typechecking, as the type of
+     * 'errorI18nKey' is a string, and the string literal cannot undergo validation.
+     */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return t(`communication-preference:error-message.${errorI18nKey}` as any);
+  }
+
+  const errorMessages = {
+    preferredMethod: getErrorMessage(actionData?.errors.preferredMethod?._errors[0]),
+    email: getErrorMessage(actionData?.errors.email?._errors[0]),
+    confirmEmail: getErrorMessage(actionData?.errors.confirmEmail?._errors[0]),
+    preferredLanguage: getErrorMessage(actionData?.errors.preferredLanguage?._errors[0]),
+    sameEmail: getErrorMessage(actionData?.errors._errors[0]),
+  };
+
+  const errorSummaryItems = createErrorSummaryItems(errorMessages);
+
+  useEffect(() => {
+    if (actionData?.formData && hasErrors(actionData.formData)) {
+      scrollAndFocusToErrorSummary(errorSummaryId);
+    }
+  }, [actionData]);
+
   return (
     <>
+      {errorSummaryItems.length > 0 && <ErrorSummary id={errorSummaryId} errors={errorSummaryItems} />}
       <p>{t('communication-preference:note')}</p>
       <Form method="post" noValidate>
-        <div className="my-6">
+        <div id="preferredMethod" className="my-6">
           {preferredCommunicationMethods.length > 0 && (
             <fieldset id="input-radios=preferred-method" className="mb-6" data-testid="input-radios-preferred-method">
```

Comment:
Use `InputRadios` with the `InputRadio`'s new prop `append`.

Reply:
updated